base/shell/cmd/cmd.c

```diff
@@ -2333,28 +2333,11 @@ static VOID Cleanup(VOID)
         ParseCommandLine(_T("\\cmdexit.bat"));
     }
 
-#ifdef FEATURE_DIRECTORY_STACK
-    /* Destroy directory stack */
-    DestroyDirectoryStack();
-#endif
-
-#ifdef FEATURE_HISTORY
-    CleanHistory();
-#endif
-
-    /* Free GetEnvVar's buffer */
-    GetEnvVar(NULL);
-
-    /* Remove ctrl break handler */
-    RemoveBreakHandler();
-
```

Comment:
Why?

Reply:
Because this ends up passing a lot of time in the heap implementation instead of just not caring like it can. The kernel frees the virtual memory when a process dies, and this is all that is needed to be done.

Reply:
OK since you now just disable the custom heap allocator by just removing the define, while keeping its code present, i.e. one can compile it back by adding the flag back; please restore these cleaning commands but surround them inside `#ifdef _DEBUG_MEM`, so that when the heap allocator is enabled, it can correctly count the cleanup there (and not telling "you've memory leaks").

Reply:
Done